doc/sphinx/addendum/extraction.rst

```diff
@@ -399,6 +399,97 @@ As an example of translation to a non-inductive datatype, let's turn
+ * which ML functions shall be accessible by the C code (callbacks)
+
+.. cmd:: Extract Foreign Constant @qualid => @string
+
+   Same as the :cmd:`Extract Constant`, except that the given ML terms
+   will be declared in the form
+
+   ``external @qualid : ML type = "@string"``.
+
+.. caution::
+
+   * The external function name provided by the user as :n:`@string`
+     is currently **not** checked at all by extraction, even for syntax errors.
+
+   * It is the responsibility of the user to ensure that the C
+     functions given to realize the axioms do have the expected or compatible
+     types. In fact, the strings containing realizing code are just copied to the
+     extracted files.
+
+.. exn:: Extract Foreign Constant is supported only for OCaml extraction
+
+   We currently only support calling foreign functions for OCaml.
+
+.. exn:: The term @qualid is already defined as custom constant.
+
+   This error message means that the :n:`@qualid` was previously used in the command
+   :cmd:`Extract Constant` or :cmd:`Extract Inlined Constant`. Using :cmd:`Extract Foreign Constant`
+   for :n:`@qualid` would override those commands.
+
+.. cmd:: Extract Callback {? @string } @qualid
+
+   This command makes sure that after extracting the :term:`constants <constant>`
+   specified by :n:`@qualid`, a constant ML function will be generated that
+   registers :n:`@qualid` as callback, callable by :n:`@string`.
+   This is done by declaring a function
+   ``let _ = Callback.register "@string" @qualid``
+   This expression signals OCaml that the given ML function :n:`@qualid` shall be
+   accessible via the alias :n:`@string`, when calling from C/C++.
+   If no alias is specified, it is set to the string representation of :n:`@qualid`.
+
+.. caution::
+   * The optional alias provided by the user as :n:`@string`
+     is currently **not** checked at all by extraction, even for syntax errors.
+
+   * It is the responsibility of the user to ensure that the callback aliases are
+     unique, i.e. when multiple modules expose a callback, the user should make
+     sure that no two :n:`@qualid` share the same alias.
+
+.. note::
+   Using Extract Callback has no impact on the rest of the synthesised code since
+   it is an additional declaration. Thus, there is also no impact on correctness
+   and type safety of the generated code.
+
+.. exn:: Extract Callback is supported only for OCaml extraction
+
+   The concept of callback registration via calling ``Callback.register`` is specific
+   for OCaml. Thus, the command is only usable when extracting OCaml code.
+
+.. cmd:: Print Extraction Foreign
+
+   Prints the current state of the set recording the custom foreigns
+   declared by the command :cmd:`Extract Foreign Constant` together with
+   the defined foreign ML function name.
+
+.. cmd:: Reset Extraction Foreign
+
+   Resets the current state of the set recording the custom externals
+   declared by the command :cmd:`Extract Foreign Constant`.
```

Comment:
```suggestion
   Resets the set of custom externals
   declared by the command :cmd:`Extract Foreign Constant`.
```

or "Clears ..."

Reply:
Resets is okay here, I think. This wording is consistent with the description of previously existing Reset commands